services/brig/src/Brig/API/User.hs

```diff
@@ -409,37 +415,38 @@ createUser new = do
 
     acceptTeamInvitation ::
       UserAccount ->
-      Team.Invitation ->
-      Team.InvitationInfo ->
+      Store.StoredInvitation ->
+      Store.StoredInvitationInfo ->
       EmailKey ->
       UserIdentity ->
       ExceptT RegisterError (AppT r) ()
-    acceptTeamInvitation account inv ii uk ident = do
+    acceptTeamInvitation account inv invitationInfo uk ident = do
       let uid = userId (accountUser account)
       ok <- lift $ liftSem $ claimKey uk uid
       unless ok $
         throwE RegisterErrorUserKeyExists
-      let minvmeta :: (Maybe (UserId, UTCTimeMillis), Role)
-          minvmeta = ((,inCreatedAt inv) <$> inCreatedBy inv, Team.inRole inv)
-      added <- lift $ liftSem $ GalleyAPIAccess.addTeamMember uid (Team.iiTeam ii) minvmeta
+      let minvmeta :: Maybe (UserId, UTCTimeMillis)
+          minvmeta = (,inv.createdAt) <$> inv.createdBy
+          role :: Role
+          role = fromMaybe defaultRole inv.role
```

Comment:
how come this wasn't here before?

Reply:
It was somewhere else, we've streamlined it.